Source/Reporters/JunitXMLReporter.jsl

```diff
@@ -0,0 +1,321 @@
+                ];
+
+  get_xml_report(xml_results);   
+  ---Text---  
+  Expected Return:
+    <testsuites failures="1" id="20191011_175324" tests="1">
+      <testsuite failures="1" name="Example Tests" tests="1">
+        <testcase name="Test1">
+          <failure type="failure">
+            Expected: add_nums(1, 1) equal to 3
+            But: was 2
+          </failure>
+        </testcase>
+      </testsuite>
+    </testsuites>
+    ---------
+  */
+
+  get_xml_report = Method({xml_results},
+    start_line = "<?xml version=\!"1.0\!" encoding=\!"UTF-8\!" ?>\!n";
+    
+    content = "";
+    test_suites = xml_results << get keys;
+    test_cases_set = xml_results << get values;
+    For(i=1, i<=NItems(test_suites), i++,
+    
+
+      test_cases = test_cases_set[i];
+      test_case_names = test_cases << get keys;
+      test_case_info = test_cases << get values;
+      test_case_xml = "";
+      For(k=1, k<=NItems(test_case_names), k++,
+
+        info_types = test_case_info[k] << get keys;
+        
+        test_case_content = "";
+        For(l=1, l<=NItems(test_case_info[k]), l++,
+          info_type = info_types[l];
+          props = Associative Array();
+          props["type"] = info_type;
+          failure_content = this:generate_xml_tag("failure", test_case_info[k][info_type], props);
+
+          test_case_content ||= failure_content;
+
+        );
+        props = Associative Array();
+        props["name"] = test_case_names[k];
+        test_case_xml ||= this:generate_xml_tag("testcase", trim(test_case_content), props, 0);
+        
+      );
+      num_tests = NItems(test_case_names);
+      props = Associative Array();
+      props["name"] = test_suites[i];
+      props["tests"] = Char(num_tests);
+
+      // find test results that are non empty
+      num_successes = Nitems(loc(test_case_info, [=>]));
+      props["failures"] = Char(num_tests-num_successes);
+      test_suite_tag = this:generate_xml_tag("testsuite", trim(test_case_xml), props, 0);
+      content ||= test_suite_tag;
+    );
+    
+    props = Associative Array();
+    date = today();
+    props["id"] = format(date, "yyyymmdd" ) || "_" || Char(Hour(date)) || Char(Minute(date)) || Char(Second(date));
+    props["tests"] = Char(Nitems(this:successes) + NItems(this:failures) + NItems(this:unexpected throws));
+    props["failures"] = Char(NItems(this:failures) + NItems(this:unexpected throws));
+
+    xml_result = start_line || this:generate_xml_tag("testsuites", trim(content), props, 0);
+    Return(xml_result);
+  );
+
+  // Method: parse_label
+  //
+  // Parses a label into test suite, test name, and case
+  //
+  // Arguments: 
+  //  label - A label separated by a separator
+  //  sep - Separator that splits the label
+  //  hash_results - Associative Array with test suites as keys and associative arrays with test cases as keys as values
+  // 
+  // Results:
+  //  The label split by the separator. The second and third values will be joined by an _
```

Comment:
Any reason not to join with `ut concat test label sep` since we already have that restriction?

Reply:
it could not recognize the variable "ut concat test label sep" unless I passed it to the method as an argument. I'm not sure why

Reply:
`Methods` are implicitly `Default Local`. So it probably thought that was a local. Try explicitly scoping it with `::ut concat test label sep`.

Reply:
That does seem to work, however I am curious as to why it is not required in the get report method